src/test/java/spoon/test/model/IncrementalBuildTest.java

```diff
@@ -0,0 +1,119 @@
+package spoon.test.model;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import spoon.Launcher;
+import spoon.SpoonModelBuilder;
+import spoon.reflect.CtModel;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+import spoon.support.SerializationModelStreamer;
+
+public class IncrementalBuildTest {
+
+	private static Factory loadFactory(File file) throws IOException {
+		return new SerializationModelStreamer().load(new FileInputStream(file));
+	}
+
+	private static void saveFactory(Factory factory, File file) throws IOException {
+		ByteArrayOutputStream outstr = new ByteArrayOutputStream();
+		new SerializationModelStreamer().save(factory, outstr);
+		OutputStream fileStream = new FileOutputStream(file);
+		outstr.writeTo(fileStream);
+	}
+
+	private static void createFile(File file, String content) throws IOException {
+		PrintWriter writer = new PrintWriter(file, "UTF-8");
+		writer.println(content);
+		writer.close();
+	}
+
+	@Test
+	public void testIncremental() throws Exception {
+		final File PROJECT_DIR = new File("./src/test/resources/inremental-test-project/");
+
+		FileUtils.deleteDirectory(PROJECT_DIR);
+		PROJECT_DIR.mkdirs();
+		createFile(new File(PROJECT_DIR, "A.java"), "public class A { int x; }");
```

Comment:
Not sure why you don't put the resources files directly in Spoon: you could copy the resources in a temp dir for tests and apply your changes on the temp dir.